src/machine/machine-dbus.c

```diff
@@ -621,23 +621,44 @@ int bus_machine_method_open_shell(sd_bus_message *message, void *userdata, sd_bu
         if (r < 0)
                 return r;
         user = empty_to_null(user);
-        if (isempty(path))
-                path = "/bin/sh";
-        if (!path_is_absolute(path))
-                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Specified path '%s' is not absolute", path);
-
-        r = sd_bus_message_read_strv(message, &args);
+        r = sd_bus_message_read_strv(message, &v);
         if (r < 0)
                 return r;
-        if (strv_isempty(args)) {
-                args = strv_free(args);
+        if (isempty(path)) {
+                char *a, *b, *get_shell;
 
-                args = strv_new(path, NULL);
+                path = "/bin/sh";
+                a = strdup("sh");
+                if (!a)
+                        return -ENOMEM;
+                b = strdup("-c");
+                if (!b)
+                        return -ENOMEM;
+                r = asprintf(&get_shell, /* This only forks one process, getent, and checks for its existance. */
```

Comment:
exist*e*nce

Reply:
The comment does not apply to the code anymore, anyway.